signed-memo/program/src/processor.rs

```diff
@@ -0,0 +1,107 @@
+//! Program state processor
+
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult, msg, program_error::ProgramError,
+    pubkey::Pubkey,
+};
+use std::str::from_utf8;
+
+/// Instruction processor
+pub fn process_instruction(
+    _program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    input: &[u8],
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let mut missing_required_signature = false;
+    for account_info in account_info_iter {
+        if let Some(address) = account_info.signer_key() {
+            msg!("Signed by {:?}", address);
+        } else {
+            msg!("Not signed! {:?}", account_info.key);
+            missing_required_signature = true;
+        }
+    }
+
+    let memo = from_utf8(input).map_err(|_| ProgramError::InvalidInstructionData)?;
+    msg!("{:?}", memo);
+
+    if missing_required_signature {
+        return Err(ProgramError::MissingRequiredSignature);
+    }
```

Comment:
I think we should bail on MissingRequiredSignature before verifying and printing the memo